packages/menu/src/utils.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { Reducer } from 'react';
+import { KEYS } from '@zendeskgarden/container-utilities';
+import { MenuItem, IMenuItemBase, MenuItemSeparator, ISelectedItem, ItemValue } from './types';
+
+export const StateChangeTypes: Record<string, string> = {
+  TriggerClick: 'trigger:click',
+  TriggerKeyDownArrowDown: `trigger:keyDown:${KEYS.DOWN}`,
+  TriggerKeyDownArrowUp: `trigger:keyDown:${KEYS.UP}`,
+  MenuKeyDownEscape: `menu:keyDown:${KEYS.ESCAPE}`,
+  MenuKeyDownTab: `menu:keyDown:${KEYS.TAB}`,
+  MenuBlur: 'menu:blur',
+  MenuMouseLeave: 'menu:mouseLeave',
+  MenuItemClick: 'menuItem:click',
+  MenuItemMouseMove: `menuItem:mouseMove`,
+  MenuItemKeyDown: 'menuItem:keyDown',
+  MenuItemKeyDownEnter: `menuItem:keyDown:${KEYS.ENTER}`,
+  MenuItemKeyDownSpace: `menuItem:keyDown:Space`,
+  MenuItemKeyDownArrowLeft: `menuItem:keyDown:${KEYS.LEFT}`,
+  MenuItemKeyDownArrowRight: `menuItem:keyDown:${KEYS.RIGHT}`,
+  MenuItemKeyDownArrowUp: `menuItem:keyDown:${KEYS.UP}`,
+  MenuItemKeyDownArrowDown: `menuItem:keyDown:${KEYS.DOWN}`,
+  MenuItemKeyDownHome: `menuItem:keyDown:${KEYS.HOME}`,
+  MenuItemKeyDownEnd: `menuItem:keyDown:${KEYS.END}`
+};
+
+export const isValidItem = (item: MenuItem) =>
+  !(item as IMenuItemBase).disabled &&
+  !(item as MenuItemSeparator).separator &&
+  !Object.hasOwn(item, 'items');
```

Comment:
```suggestion
  !isItemGroup(item)
```

Does this read better?